streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/SuppressedImpl.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals.suppress;
+
+import org.apache.kafka.streams.kstream.Suppressed;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.processor.ProcessorContext;
+
+import java.time.Duration;
+
+import static org.apache.kafka.streams.kstream.internals.suppress.SuppressedImpl.BufferFullStrategy.SHUT_DOWN;
+import static org.apache.kafka.streams.kstream.internals.suppress.SuppressedImpl.BufferFullStrategy.SPILL_TO_DISK;
+
+public final class SuppressedImpl<K, V> implements Suppressed<K, V> {
+    private IntermediateSuppressionImpl<K, V> intermediateSuppression = null;
+    private TimeDefinition<K, V> timeDefinition = (context, k, v) -> context.timestamp();
+    private StrictBufferConfig<K, V> finalResultsConfig;
+
+    enum BufferFullStrategy {
+        EMIT,
+        SPILL_TO_DISK,
+        SHUT_DOWN
+    }
+
+    public static class StrictBCImpl<K, V> extends AbstractBufferConfig<K, V, StrictBufferConfig<K, V>> implements StrictBufferConfig<K, V> {
+
+        private final long maxKeys;
+        private final long maxBytes;
+        private final BufferFullStrategy bufferFullStrategy;
+
+        StrictBCImpl(final long maxKeys,
```

Comment:
Could this be `private`?

Reply:
Hmm, I'll check. I don't think so, but I don't remember why.

Reply:
It is used in the static factory method of the interface.